src/tools/periph/mkfatimage16.c

```diff
@@ -467,24 +467,31 @@ int main(int argc, char *argv[])
   bpb->num_heads = heads;
 
   if (bootsect_file) {
-    switch (bpb->v340.signature) {
-      case 0x29: // v400
+    if(bpb->v340.signature == 0x28 || bpb->v340.signature == 0x29) {
+      if(bpb->v340.signature == 0x29) { // v400
         memset(bpb->v400.vol_label, ' ', 11);
         memcpy(bpb->v400.vol_label, volume_label, strlen(volume_label));
         memcpy(bpb->v400.fat_type,
                p_type == P_TYPE_12BIT ? "FAT12   " : "FAT16   ", 8);
-        /* fall through */
-      case 0x28: // v340
-        bpb->v340.drive_number = 0x80;
-        bpb->v340.flags = 0;
-        bpb->v340.serial_number = 0x12345678;
-        /* fall through */
-      default: //  v331 compatible
-        if (p_sectors >= 65536) { // Only rewrite if we are large
-          bpb->v331.hidden_sectors = HIDDEN_SECTORS;
-          bpb->v331.num_sectors_large = p_sectors;
-        }
-        break;
+      }
+      // v400 && v340
+      bpb->v340.drive_number = 0x80;
+      bpb->v340.flags = 0;
+      bpb->v340.serial_number = 0x12345678;
```

Comment:
We are only making an image here, so are writing reasonable defaults. When using it later in dosemu it may be fixed up when we know what values to assign, e.g. for drive number etc.